oauth.rst

```diff
@@ -1,131 +1,129 @@
 
-The Resource Server
--------------------
-What you would typically think of as the main API is called the "resource server". In our case this is the ``dockapi``: the server that contains the user's information that is being accessed by the third-party application.
 
-The Authorization Server
-------------------------
-The authorization server is what the user interacts with when an application is requesting access to their account. In our case this is the ``dock-auth`` service.
+Basic Client Integration Flow
+=============================
 
-The Client
-----------
-The Client is the 3rd party app that is attempting to act on the user's behalf, or access the user's resources. Before the Client can access the user's account, it needs to obtain permission from the user.
+Quick details
+-------------
+Following is the simplified version of the flow before diving into details: 
 
-Client Registration
--------------------
-When developers register their applications they need to provide some basic info:
+- Clients/Partner request an ``authorization_code`` grant type by redirecting the user to the Dock Authorization Page. 
+- The user verifies the information the Client app is requesting access to (the scopes) and also see the app's name, logo and description. 
+- After approval, the user is redirected to the given ``return_uri`` with an ``authorization code`` added as a parameter. 
+- The client can then do a backend call (not redirection) to exchange it for an ``access token``. 
+- Dock will only return an ``access token`` to the right ``authorization code`` coming from the right Client, using the right client credentials. 
+- Finally, by using the Bearer Token in the ``Authorization`` header, the client will be able to access the requested user data.
 
-- App name
-- App logo
-- App description
+Detailed information
+--------------------
+To make sense of the above it can be useful to observe the whole flow in a detailed step-by-step explanation. The following steps explain how the flow would look like for a Client getting an Access Token to act on behalf of one of its users, and then using it to get user data from Dock:
 
-And also some URLs:
+Step 1 - Authorization grant query
+----------------------------------
+To initiate the OAuth flow the Client needs to start an authorization grant query by redirecting the user to:
+``GET https://app.dock.io/oauth/authorize`` with the following query params:
 
-- Homepage url
-- Privacy policy url
-- Callback URIs (this is a list)
+- ``client_id``: id of the Client doing the request. This is the id you received when you registered your application.
+- ``response_type=code``: 'code' is always expected, it means that you expect to get an ``authorization_code``
```

Comment:
I'd change this to "``response_type``: 'code' is always expected, it means that the client expects to get an ``authorization_code`` back."